index.d.ts

```diff
@@ -0,0 +1,18 @@
+import { cleanup, act, RenderOptions, RenderResult } from 'react-testing-library'
+
+export function renderHook<P extends any, R extends any>(
+  callback: (...args: [P]) => R,
```

Comment:
is the `...args: [P]` saying that there can be more than 1 arg, or is it some syntax to describe the first (and only) arg?

Reply:
The latter is true.
We can use *Tuple type* to extract args types.

```js
function foo<A1, A2, A3>(...args: [A1, A2, A3]): { a1: A1, a2: A2, a3: A3 } {
    return { a1: args[0], a2: args[1], a3: args[2] };
}

foo(1, '', true); // => the return value type is `{ a1: number, a2: string, a3: boolean }`
```

https://www.typescriptlang.org/play/index.html#src=function%20foo%3CA1%2C%20A2%2C%20A3%3E(...args%3A%20%5BA1%2C%20A2%2C%20A3%5D)%3A%20%7B%20a1%3A%20A1%2C%20a2%3A%20A2%2C%20a3%3A%20A3%20%7D%20%7B%0D%0A%20%20%20%20return%20%7B%20a1%3A%20args%5B0%5D%2C%20a2%3A%20args%5B1%5D%2C%20a3%3A%20args%5B2%5D%20%7D%3B%0D%0A%7D%0D%0A%0D%0Aconst%20res%3A%20%7B%20a1%3A%20number%2C%20a2%3A%20string%2C%20a3%3A%20boolean%20%7D%20%3D%20foo(1%2C%20''%2C%20true)%3B

Reply:
I used `[P]` to extract first argment type here.

Reply:
If you want just a first arg, shouldn't it work like this as well? I am wondering what's the difference?
```js
  callback: (props: P) => R,
```

Reply:
That's absolutely right.
Thanks for your suggestion!